nexus/api.go

```diff
@@ -176,3 +201,107 @@ func addContextTimeoutToHTTPHeader(ctx context.Context, httpHeader http.Header)
 	httpHeader.Set(HeaderRequestTimeout, time.Until(deadline).String())
 	return httpHeader
 }
+
+// Link contains an URL and a Type that can be used to decode the URL.
+// Links can contain any arbitrary information as a percent-encoded URL.
+// It can be used to pass information about the caller to the handler, or vice-versa.
+type Link struct {
+	// URL information about the link.
+	// It must be URL percent-encoded.
+	URL string
+	// Type can describe an actual data type for decoding the URL.
+	// Valid chars: alphanumeric, '_', '.', '/'
+	Type string
+}
+
+const linkTypeKey = "type"
+
+func encodeLink(link Link) (string, error) {
+	if err := validateLinkURL(link.URL); err != nil {
+		return "", fmt.Errorf("failed to encode link: %w", err)
+	}
+	if err := validateLinkType(link.Type); err != nil {
+		return "", fmt.Errorf("failed to encode link: %w", err)
+	}
+	return fmt.Sprintf(`<%s>; %s="%s"`, link.URL, linkTypeKey, link.Type), nil
+}
+
+func decodeLink(encodedLink string) (Link, error) {
+	var link Link
+	parts := strings.Split(encodedLink, ";")
+
+	u := strings.TrimSpace(parts[0])
+	if !strings.HasPrefix(u, "<") || !strings.HasSuffix(u, ">") {
+		return link, fmt.Errorf("failed to parse link header value: %s", encodedLink)
```

Comment:
Maybe add some context?

```suggestion
		return link, fmt.Errorf("failed to parse link header value: %q, expected URL to be enclosed between < and >", encodedLink)
```